opensrs/opensrsapi.py

```diff
@@ -405,6 +411,8 @@ def register_domain(self, domain, purchase_period, user, user_id,
                             'Invalid syntax on domain')):
                     raise errors.InvalidDomain(e)
                 raise errors.DomainRegistrationFailure(e)
+            if e.response_code == self.CODE_CLIENT_TIMED_OUT:
+                raise errors.DomainRegistrationTimedOut(e)
```

Comment:
And to capture those cases, I think we should start with adding some better logging in DS (probably in `translate_errors` decorator).  Since for now we just hide all those external errors under our generic `UnknownServiceError`.